src/apify/scrapy/cache.py

```diff
@@ -0,0 +1,176 @@
+from __future__ import annotations
+
+import gzip
+import io
+import pickle
+import struct
+from logging import getLogger
+from time import time
+from typing import TYPE_CHECKING
+
+from scrapy.http.headers import Headers
+from scrapy.responsetypes import responsetypes
+
+from ._async_thread import AsyncThread
+from apify import Configuration
+from apify.apify_storage_client import ApifyStorageClient
+from apify.storages import KeyValueStore
+
+if TYPE_CHECKING:
+    from scrapy import Request, Spider
+    from scrapy.http.response import Response
+    from scrapy.settings import BaseSettings
+    from scrapy.utils.request import RequestFingerprinterProtocol
+
+logger = getLogger(__name__)
+
+
+class ApifyCacheStorage:
+    """A Scrapy cache storage that uses the Apify `KeyValueStore` to store responses.
+
```

Comment:
Could you add that it is being used by the Scrapy's `HttpCacheMiddleware`?

Reply:
Addressed in 155caf2